### PR TITLE
Fix list scatter child indexing for sliced inputs

### LINKED ISSUE #23818: List scatter misindexes child rows for sliced sources and targets
Scattering into or from a sliced lists column produces wrong child rows whenever the list child is not fixed-width.

`list_child_constructor` in `cpp/src/lists/copying/scatter_helper.cu` reads the parent lists offsets with the raw row index and ignores the column view offset:

- strings branch: `strings_offset = lists_offsets_ptr[row_index] + intra_index`
- nested-lists branch: `child_row_index = lists_offsets_ptr[row_index] + intra_index`

For a sliced source or target view, `row_index` is relative to the view but the offsets data pointer is not shifted, so both reads land on the wrong offsets entry and the constructed child copies from wrong rows of the underlying buffer. The fixed-width branch is unaffected because element access goes through `list_device_view`, whose constructor already applies the offset (`offsets.element(row_index + lists_column.offset())`).

The struct branch has the same problem one level up: `project_member_as_list_view` builds the member-as-list `column_view` with the offset hardcoded to `0`, so a sliced `List<Struct>` source or target misindexes both offsets values and null-mask bits inside its members.

Observed with a standalone program against libcudf 26.10 nightly: scattering a sliced 4-row `List<List<int>>` source (slice {1,3}) produced leaf values `10 11 12 13 14` where `13 14 15` was expected.

Suggested fix: use `lists_column_device_view::offset_at(row_index)` (which indexes `offsets()[offset() + idx]`) at the two device sites, and pass the parent's `offset()` into `project_member_as_list_view`. Both are no-ops when `offset() == 0`.

### PR DESCRIPTION
**Pull request by VaggelisGian**
## Description

Scattering into or from a **sliced** lists column produced wrong child rows whenever the list child is strings or another list: `list_child_constructor` read the parent offsets with the raw row index and ignored the column view offset.

Concretely, for a sliced source `List<List<int>>` the scattered rows picked up leaf values from earlier rows of the underlying child buffer (observed `10 11 12 13 14` where `13 14 15` was expected), and a sliced `List<string>` source resolves string positions against the wrong offsets entries.

The fixed-width branch was unaffected because element access goes through `list_device_view`, whose constructor already applies the view offset (`offsets.element(row_index + lists_column.offset())`). The strings and nested-lists branches read the raw offsets pointer directly and missed that arithmetic. The struct branch had the same defect one level up: `project_member_as_list_view` hardcoded the projected member view's offset to `0`, so sliced `List<Struct>` inputs misindexed offsets values and null-mask bits inside members.

Changes:
- strings and nested-lists branches use `lists_column_device_view::offset_at(row_index)`, which is exactly the canonical accessor (`offsets().element(offset() + idx)`).
- `project_member_as_list_view` now forwards the parent's `offset()` into the synthetic member-as-list view.
- All three changes are bit-identical no-ops when `offset() == 0`.

Fixes #23818

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/copying/scatter_list_tests.cpp [hunk 1]
```diff
@@ -75,6 +75,69 @@ TYPED_TEST(TypedScatterListsTest, SlicedInputLists)
     cudf::test::lists_column_wrapper<T, int32_t>{{8, 8, 8}, {2, 2}, {9, 9, 9, 9}, {4, 4}, {5, 5}});
 }
 
+TYPED_TEST(TypedScatterListsTest, SlicedInputListsOfLists)
```
Comment (wence-) [inline cpp/tests/copying/scatter_list_tests.cpp:78]: question: Can we also test the case where the source has a null mask? I think we can do it in this test by slicing from `{1, 4}` and having one of the rows we scatter be null.

Same, mutatis mutandis, in the other tests.

**Review by wence- — CHANGES_REQUESTED** [review]

File: cpp/tests/copying/scatter_list_tests.cpp (hunk 1 above)
Comment (VaggelisGian) [reply cpp/tests/copying/scatter_list_tests.cpp:78]: Pushed 22e488949a: all three sliced-input tests now scatter one null source row and expect the corresponding output row to stay null.

- SlicedInputListsOfLists and SlicedInputListsOfStrings keep the {1, ...} slicing pattern with the null list row at source index 2 so it lands inside the sliced view that gets scattered.
- SlicedInputListsOfStructs carries its null mask through detail::make_null_mask since it builds the column with make_lists_column directly.

Not compiled locally, no Linux toolchain on this machine. Could you /ok to test 22e488949a when convenient?
